File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxOnes = 0 ;
        int consOnes = 0 ; 
        for(int i = 0 ; i < nums.size() ; i++){
            if(nums[i]==1){
                consOnes++;
                if(consOnes > maxOnes ) {
                    maxOnes = consOnes ; 
                }
            }
            else {
                consOnes = 0 ; 
            }
        
        }
        return maxOnes ; 
    }
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cs
public class Solution {
    public int FindMaxConsecutiveOnes(int[] nums) {
        int maxOnes = 0 , currentMax = 0 ; 
        for(int i = 0 ; i < nums.Length ; i++){
            if(nums[i] == 1){
                currentMax++;
                if(currentMax > maxOnes){
                    maxOnes = currentMax ;
                }
            }
            else
                currentMax = 0 ;
        }
        return maxOnes;
    }
}